backend/app/donation/routes.py

```diff
@@ -0,0 +1,100 @@
+from flask import Flask, jsonify, request, abort, make_response
+from . import donation
+from .. import db
+from app.models import Donation
+import uuid
+from datetime import datetime
+
+# Get all donations of user_id
+@donation.route("/donations/<user_uuid>", methods=["GET"])
+def get_denoations_for_user(user_uuid):
+    # Look for all donations of "user_id"
+    donations = Donation.query.filter_by(added_by=user_uuid).all()
+    return jsonify(Donation.serialize_list(donations))
+
+
+# Create new donation
+@donation.route("/create-donation", methods=["POST"])
+def create_donation():
+    # Load request to data
+    data = request.get_json(force=True)
+    # Assign id new uuid, and set date to now
+    id = uuid.uuid4()
+    date = datetime.now()
+    # Load data to the corresponding variables
+    name = data.get("name")
+    email = data.get("email")
+    donation_source = data.get("donation_source")
+    event = data.get("event")
+    num_tickets = data.get("num_tickets")
+    added_by = data.get("added_by")
+    # Check if nullable=False columns are empty
+    if name == "" or donation_source == "" or email == "":
```

Comment:
@ahmad675 I was wondering whether you could write a unit test to test this?

Reply:
https://github.com/hack4impact-mcgill/mu-crm-tool/pull/65/files#r566335911 same issue here where we should be checking for`{}` empty object